From 71dd5633e085fb1d651e1fba55bc8ae51327b8d5 Mon Sep 17 00:00:00 2001
From: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
Date: Tue, 24 Sep 2013 10:53:11 +0300
Subject: [PORT FROM MAIN] Telephony: configure retry alarm only if data is allowed

BZ: 106926

Currently, reconnect alarm is getting configured even
when the data is not allowed. Reconnect alarm is 5seconds.
If the user enables the data again just when the alarm goes
off, SETUP_DATA_CALL will get triggered with a Data connection
channel which will get reset to null on retry intent handling.
Even if the SETUP_DATA_CALL succeeds, framework considers
this as an due to no data channel associated with the apnContext.
So, data will be deactivated and won't be retried again.

Configure the retry alarm only if the data is enabld and
allowed.

Change-Id: I8f4f02e3a37dea020d08b31cd77e4cf9b6c448fa
Orig-Change-Id: Ib8f0e18eb3eabc187af41cbb2e568b9823e93395
Category: aosp improvement
Domain: Telephony-Telephony-FWK
Origin: internal
Upstream-Candidate: yes
Signed-off-by: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
Signed-off-by: Arun Ravindran <arun.ravindran@intel.com>
---
 .../telephony/dataconnection/DcTracker.java        |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/java/com/android/internal/telephony/dataconnection/DcTracker.java b/src/java/com/android/internal/telephony/dataconnection/DcTracker.java
index 8ab80dd..d001eff 100644
--- a/src/java/com/android/internal/telephony/dataconnection/DcTracker.java
+++ b/src/java/com/android/internal/telephony/dataconnection/DcTracker.java
@@ -1828,7 +1828,7 @@ public final class DcTracker extends DcTrackerBase {
         }
 
         // If APN is still enabled, try to bring it back up automatically
-        if (mAttached.get() && apnContext.isReady() && retryAfterDisconnected(apnContext)) {
+        if (mAttached.get() && isDataAllowed(apnContext) && retryAfterDisconnected(apnContext)) {
             SystemProperties.set(PUPPET_MASTER_RADIO_STRESS_TEST, "false");
             // Wait a bit before trying the next APN, so that
             // we're not tying up the RIL command channel.
-- 
1.7.4.1

